modules/formatter/src/ast/node.scala

```diff
@@ -0,0 +1,196 @@
+
+object QuotedChar {
+
+  case class SimpleCharCase(char: Char) extends QuotedChar
+
+  case class EscapedCharCase(char: EscapedChar) extends QuotedChar
+
+  case class PreservedDoubleCase(preservedDouble: PreservedDouble)
+      extends QuotedChar
+
+  case object NewLineCase extends QuotedChar
+}
+
+case class EscapedCharValue(escapedChar: EscapedChar)
+
+//    escape (escape / "'" / DQUOTE / "b" / "f" / "n" / "r" / "t" / "/" / unicode_escape)
+sealed trait EscapedChar
+
+object EscapedChar {
+  case class CharCase(char: Char) extends EscapedChar
+
+  case class UnicodeEscape(hex: Char, hex2: Char, hex3: Char)
+      extends EscapedChar
+}
+
+case class PreservedDouble(char: Char)
+
+case class TextBlock(quotedChars: List[QuotedChar])
+
+object node_parser {
+
+  val opParser: Parser[Char] = Parser.charIn(op)
+  val qChar: Parser[Char] = Parser.charIn(allQuotable)
+  val preserved_double: Parser[PreservedDouble] = escape *> qChar
+    .map(PreservedDouble)
+  val unicode_escape: Parser[UnicodeEscape] =
+    (Parser.char('u') *> hexdig ~ hexdig ~ hexdig).map { case ((a, b), c) =>
+      UnicodeEscape(a, b, c)
+    }
+
+  val escaped_char: Parser[EscapedChar] =
+    escape *> (Parser.charIn(escapeChars).map(CharCase) | unicode_escape)
+  val QuotedChar: Parser[QuotedChar] =
+    qChar.backtrack.map(SimpleCharCase) | escaped_char.backtrack.map(
+      EscapedCharCase
+    ) | preserved_double.map(PreservedDoubleCase) | NL.as(NewLineCase)
+
+  val ThreeDquotes = dquote ~ dquote ~ dquote
+  val text_block: Parser[TextBlock] =
+    ((ThreeDquotes ~ sp0 *> NL) *> QuotedChar.rep0 <* ThreeDquotes)
+      .map(TextBlock)
+  val quoted_text: Parser[QuotedText] =
+    (dquote *> QuotedChar.rep0 <* dquote).map(QuotedText)
+  val node_string_value: Parser[NodeStringValue] = shape_id.backtrack.map(
+    ShapeIdCase
+  ) | text_block.backtrack.map(TextBlockCase) | quoted_text.backtrack
+    .map(QuotedTextCase)
+  val node_keywords: Parser[NodeKeyword] =
+    Parser.stringIn(Set("true", "false", "null")).map(NodeKeyword)
+
+  val frac: Parser[Frac] = Parser.char('.') *> digit.string.map(Frac)
+  val exp: Parser[Exp] =
+    (Parser.charWhere(c => c == 'e' || c == 'E') ~ opParser.? ~ digit.string)
+      .map { case ((a, b), c) =>
+        Exp(a, b, c)
+      }
+
+  val number: Parser[SmithyNumber] =
+    (Parser.charWhere(_ == '-').?.with1 ~ nonNegativeIntString ~ frac.? ~ exp.?)
+      .map { case (((a, b), c), d) =>
+        SmithyNumber(a, b, c, d)
+      }
+  lazy val node_value: Parser[NodeValue] = Parser.defer(
+    nodeArray.backtrack | nodeObject.backtrack | number.backtrack | node_keywords.backtrack | node_string_value.backtrack
+  )
+
+  val node_object_key: Parser[NodeObjectKey] = quoted_text.backtrack
+    .map(
+      NodeObjectKey.QuotedTextNok
+    ) | identifier.map(NodeObjectKey.IdentifierNok)
+
+  lazy val node_object_kvp: Parser[NodeObjectKeyValuePair] =
+    ((node_object_key ~ ws <* Parser.char(':')) ~ ws ~ node_value).map {
```

Comment:
I'd like it if we could move all instances of `Parser.char(/*something*/)` to the `modules/formatter/src/ast/package.scala` where others are defined

IMO, it cleans up the code quite a bit, especially when the formatter comes in and add a line in there

Reply:
all parsers are going to be under the parsers package .  there is some stuff left over , i will clean up